assets/src/block-validation/sidebar.js

```diff
@@ -0,0 +1,162 @@
+
+		if ( isShowingReviewed ) {
+			updatedDisplayedErrors = updatedValidationErrors;
+		} else {
+			updatedDisplayedErrors = updatedUnreviewedValidationErrors;
+
+			// If there are no unreviewed errors, we show the reviewed errors.
+			if ( 0 === updatedDisplayedErrors.length ) {
+				updatedDisplayedErrors = updatedReviewedValidationErrors;
+			}
+		}
+
+		return {
+			displayedErrors: updatedDisplayedErrors,
+			reviewedValidationErrors: updatedReviewedValidationErrors,
+			unreviewedValidationErrors: updatedUnreviewedValidationErrors,
+			validationErrors: updatedValidationErrors,
+		};
+	}, [ isShowingReviewed ] );
+
+	/**
+	 * Focus the first focusable element when the sidebar opens.
+	 */
+	useEffect( () => {
+		const element = document.querySelector( '.amp-sidebar a, .amp-sidebar button, .amp-sidebar input' );
+		if ( element ) {
+			element.focus();
+		}
+	}, [] );
+
+	const saved = 'auto-draft' !== status;
+
+	return (
+		<div className="amp-sidebar">
+			{
+				ampCompatibilityBroken && (
+					<div className="amp-sidebar__broken-container">
+						<div className="amp-sidebar__broken">
+							<div className="amp-sidebar__validation-errors-kept-icon">
+								<AMPValidationErrorsKeptIcon />
+							</div>
+							<div>
+								<h3>
+									{ __( 'Validation errors kept', 'amp' ) }
+								</h3>
+								{ __( 'This URL is not fully AMP-compatible.', 'amp' ) }
+							</div>
+						</div>
+					</div>
+				)
+			}
+			{ 0 < validationErrors.length && (
+				<PanelBody opened={ true } className="amp-sidebar__description-panel">
+					<div className="amp-sidebar__validation-errors-icon">
+						<AMPValidationErrorsIcon />
+					</div>
+					<div className="amp-sidebar__validation-errors-heading">
+						<h2>
+							{ __( 'Validation Errors', 'amp' ) }
+						</h2>
+
+						<p>
+							{ reviewLink && (
+								<a href={ reviewLink } className="amp-sidebar__review-link" target="_blank" rel="noreferrer">
+									{ __( 'Review errors', 'amp' ) }
+									<NewTabIcon />
+								</a>
+							) }
+						</p>
+					</div>
+					{ ( 0 < reviewedValidationErrors.length && 0 < unreviewedValidationErrors.length ) && (
+						<div className="amp-sidebar__options">
+							<ToggleControl
+								checked={ isShowingReviewed }
+								label={ __( 'Include reviewed errors', 'amp' ) }
+								onChange={ ( newIsShowingReviewed ) => {
+									setIsShowingReviewed( newIsShowingReviewed );
+								} }
+							/>
+						</div>
+					) }
+				</PanelBody>
+			) }
+
+			{
+				! saved && (
+					<PanelBody opened={ true }>
+						<p>
+							{ __( 'This URL will be checked for validation errors when the post is saved.', 'amp' ) }
+						</p>
+					</PanelBody>
+				)
+			}
+			{ saved && validationErrors.length === 0 && (
+				<PanelBody opened={ true }>
+					<p>
+						{ __( 'There are no AMP validation errors at this post\'s URL.', 'amp' ) }
```

Comment:
Given it may not be a post but a page or some other CPT.

```suggestion
						{ __( 'There are no AMP validation issues at the permalink.', 'amp' ) }
```

Reply:
If we want to include "post" then we should obtain the `singular_name` from the post type.